resource-aggregate/cqrs/eventstore/cqldb/loadDeviceMetadataByServiceIDs.go

```diff
@@ -0,0 +1,55 @@
+package cqldb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/gocql/gocql"
+	"github.com/plgd-dev/hub/v2/pkg/cqldb"
+	pkgStrings "github.com/plgd-dev/hub/v2/pkg/strings"
+	"github.com/plgd-dev/hub/v2/resource-aggregate/cqrs/eventstore"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (s *EventStore) LoadDeviceMetadataByServiceIDs(ctx context.Context, serviceIDs []string, limit int64) ([]eventstore.DeviceDocumentMetadata, error) {
+	if len(serviceIDs) == 0 {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid serviceIDs")
+	}
+	serviceIDs = pkgStrings.Unique(serviceIDs)
+	q := cqldb.SelectCommand + " " + deviceIDKey + "," + serviceIDKey + " " + cqldb.FromClause + " " + s.Table() + " " + cqldb.WhereClause + " " + serviceIDKey + " in (" + strings.Join(serviceIDs, ",") + ") LIMIT " + fmt.Sprintf("%v", limit) + ";"
+	iter := s.client.Session().Query(q).WithContext(ctx).Iter()
+	if iter == nil {
+		return nil, errors.New("cannot create iterator")
+	}
+	ret := make([]eventstore.DeviceDocumentMetadata, 0, iter.NumRows())
+	values := make(map[string]interface{}, 2)
+	for iter.MapScan(values) {
+		deviceID, ok := values[deviceIDKey].(gocql.UUID)
+		if !ok {
+			_ = iter.Close()
+			return nil, fmt.Errorf(errFmtDataIsNotStringType, deviceIDKey, values[deviceIDKey])
```

Comment:
errFmtDataIsNotStringType -> is not a `gocql.UUID`

Reply:
fixed